### PR TITLE
ATV-77 Document metadata API

### PR DESCRIPTION
**Pull request by AntiAki-M**
API for fetching documents specific to a single user without including sensitive information
Tests for API
Updated docs

## Description :sparkles:
This API allows filtering through documents of a single user throughout all services without revealing sensitive information withing the documents to allow easier collaboration of different services. 

### Closes :no_good_woman:
**[ATV-77](https://helsinkisolutionoffice.atlassian.net/jira/software/c/projects/ATV/boards/207?modal=detail&selectedIssue=ATV-77): As a user I want to view the metadata of my documents in across services** 

### Automated tests :gear:️
Yes

### CODE REVIEW

File: documents/api/viewsets.py [hunk 1]
```diff
@@ -30,10 +31,70 @@
...
+from .filtersets import DocumentFilterSet, DocumentMetadataFilterSet
+from .querysets import (
+    get_attachment_queryset,
+    get_document_metadata_queryset,
+    get_document_queryset,
+)
+
+
+@extend_schema_view(**document_metadata_viewset_docs)
+class DocumentMetadataViewSet(AuditLoggingModelViewSet):
```
Comment (vinikvuo) [inline documents/api/viewsets.py:50]: Is this AuditLoggingModelViewSet own code? Could it be GenericViewSet? Not_allowing everything is kind of ugly.

File: documents/api/viewsets.py [hunk 2]
```diff
@@ -30,10 +31,70 @@
...
+    serializer_class = DocumentMetadataSerializer
+    queryset = Document.objects.none()
+    filter_backends = [DjangoFilterBackend]
+    lookup_field = "user__uuid"
+    pagination_class = PageNumberPagination
+    filterset_class = DocumentMetadataFilterSet
+
+    def get_queryset(self):
+        super().get_queryset()
```
Comment (vinikvuo) [inline documents/api/viewsets.py:59]: Is this obsolete row? Or what this is supposed to do?

Comment (AntiAki-M) [reply documents/api/viewsets.py:59]: It's useless, well spotted. Thank you! I'll fix it.

File: documents/api/viewsets.py (hunk 1 above)
Comment (AntiAki-M) [reply documents/api/viewsets.py:50]: AuditLoggingModelViewSet is part of the project. It overrides all those unallowed methods to include logging functionality with API calls. This solution also follows the code style of other previously implemented ViewSets.

File: documents/serializers/document.py [hunk 3]
```diff
@@ -17,6 +17,37 @@
 from .status_history import StatusHistorySerializer
 
 
+def status_to_representation(representation):
```
[the remark this answers is not in the document]
Comment (AntiAki-M) [reply documents/serializers/document.py:20]: This feature is a subject for change, and will be reworked soon. Currently the status field is constructed from two fields of the model, but status timestamp will become its own value afterwards. Anyways I found a bug thanks to this comment, so thank you!

**Pull request merged**